### ISSUE TITLE
New site breakdown for half-hourly checkbox breaks detailedForecast and detailedHourly

### ISSUE THREAD

**Issue by Cycor** [BJReplay/ha-solcast-solar#189]
**Describe the bug**
When I enable site breakdown I don't get any detailedForecast or detailedHourly attributes anymore

**To Reproduce**
1. Enable 'Enable site breakdown for half-hourly/hourly detail attributes (these will not be saved to database state attributes)' checkbox
4. See missing data

**Expected behavior**
Get detailedForecast and detailedHourly attributes for my 2 sites

**Logs**

```
2024-10-18 04:26:26.110 ERROR (MainThread) [custom_components.solcast_solar.sensor] Unable to get sensor value: tuple index out of range: Traceback (most recent call last):
File "/config/custom_components/solcast_solar/solcastapi.py", line 1312, in build_hourly
x1 = round((t[index]["pv_estimate"] + t[index+1]["pv_estimate"]) / 2, 4)
~^^^^^^^
IndexError: tuple index out of range
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/config/custom_components/solcast_solar/sensor.py", line 405, in extra_state_attributes
return self._coordinator.get_sensor_extra_attributes(self.entity_description.key)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/solcast_solar/coordinator.py", line 409, in get_sensor_extra_attributes
ret = self.solcast.get_forecast_day(6)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/solcast_solar/solcastapi.py", line 1357, in get_forecast_day
hourlytups[site['resource_id']] = build_hourly(tups[site['resource_id']])
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/solcast_solar/solcastapi.py", line 1317, in build_hourly
x1 = round((t[index]["pv_estimate"]), 4)
~^^^^^^^
IndexError: tuple index out of range
```

**Solcast Integration Version**
 - 4.2.0

**Desktop (please complete the following information):**
 - OS: Windows 11
 - Browser: Chrome
 - Version: 1.70.126

**Comment by BJReplay** [comment]
It's possible that you need to wait for the service to gather a forecast - though it is pretty good at working out if a config change requires it to re-fetch a forecast.  However, without debug logs, we can't tell if it tried (failed, perhaps because the Solcast servers were busy), and this error message was the end result.

Please enable debug logs as described in #38 and paste the results back here.

Please enclose all logs in the code wrapper - you can do that by highlighting the logs that you've pasted in, then clicking on the code shortcut on the github formatting toolbar, or by putting three backticks ``` before and after the logs

![image](https://github.com/user-attachments/assets/259ab3c2-9b10-4344-9e8b-e045cc6b5cce)

**Comment by autoSteve** [comment]
I have identified an issue in the code @Cycor. This will be fixed in v4.2.1.

Thanks for bringing this to our attention!

If you do not need the hourly breakdown then disabling that attribute when the detailed breakdown is enabled will work around the issue.
